engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/SchemaRegistryClient.scala

```diff
@@ -0,0 +1,25 @@
+package pl.touk.nussknacker.engine.avro
+
+import io.confluent.kafka.schemaregistry.client.SchemaMetadata
+import org.apache.avro.Schema
+import pl.touk.nussknacker.engine.kafka.KafkaConfig
+
+trait SchemaRegistryClient extends Serializable {
+
+  def getById(id: Int): Schema
+
+  def getSchemaMetadata(subject: String, version: Int): SchemaMetadata
+
+  def getLatestSchemaMetadata(name: String): SchemaMetadata
+
+  def schemaMetadata(subject: String, version: Option[Int]): SchemaMetadata =
+    version
+      .map(ver => getSchemaMetadata(subject, ver))
+      .getOrElse(getLatestSchemaMetadata(subject))
+}
+
+//It's only for internal providers usage..
+trait SchemaRegistryClientFactory[Client] extends Serializable {
```

Comment:
If it is only internal why not to make `Client` as a type `Parameter` as below? Also not sure why this `TypedSchemaRegistryClient` is necessary at all. You can always define `def createSchemaRegistryClient(kafkaConfig: KafkaConfig): SchemaRegistryClient with Client` and in implementation override it with `def createSchemaRegistryClient(kafkaConfig: KafkaConfig): ConcreteClass`

Reply:
I've provided:
```

trait ConfluentSchemaRegistryClientFactory extends SchemaRegistryClientFactory {
  override def createSchemaRegistryClient(kafkaConfig: KafkaConfig): ConfluentSchemaRegistryClient
}